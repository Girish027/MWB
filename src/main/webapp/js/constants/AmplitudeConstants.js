
const amplitudeConstants = {

  // ToolId for Modeling Work Bench
  TOOL_ID: 'MWB',

  // List of Events
  LOGIN_EVENT: 'LoginUser',
  LOGOUT_EVENT: 'LogoutUser',
  SELECT_CLIENT_EVENT: 'SelectClientApp',
  CREATE_DATASET_EVENT: 'CreateDataset',
  CREATE_DATASET_COMPLETED_EVENT: 'CreateDatasetCompleted',
  CREATE_DATASET_FAILED_EVENT: 'CreateDatasetFailed',
  DELETE_DATASET_EVENT: 'DeleteDataset',
  DELETE_DATASET_COMPLETED_EVENT: 'DeleteDatasetCompleted',
  DELETE_DATASET_FAILED_EVENT: 'DeleteDatasetFailed',
  CREATE_MODEL_EVENT: 'CreateModel',
  CREATE_MODEL_COMPLETED_EVENT: 'CreateModelCompleted',
  CREATE_MODEL_FAILED_EVENT: 'CreateModelFailed',
  DELETE_MODEL_EVENT: 'DeleteModel',
  DELETE_MODEL_COMPLETED_EVENT: 'DeleteModelCompleted',
  DELETE_MODEL_FAILED_EVENT: 'DeleteModelFailed',
  TEST_SPEECH_MODEL_EVENT: 'TestSpeechModel',
  TEST_SPEECH_MODEL_COMPLETED_EVENT: 'TestSpeechModelCompleted',
  TEST_SPEECH_MODEL_FAILED_EVENT: 'TestSpeechModelFailed',
  TEST_DIGITAL_MODEL_EVENT: 'TestDigitalModel',
  TEST_DIGITAL_MODEL_COMPLETED_EVENT: 'TestDigitalModelCompleted',
  DIGITAL_MODEL_TEST_FAILED_EVENT: 'TestDigitalModelFailed',
  DOWNLOAD_MODEL_EVENT: 'DownloadModel',
  IMPORT_TAGGING_GUIDE_EVENT: 'ImportTaggingGuide',
  IMPORT_TAGGING_GUIDE_COMPLETED_EVENT: 'ImportTaggingGuideCompleted',
  IMPORT_TAGGING_GUIDE_FAILED_EVENT: 'ImportTaggingGuideFailed',
  RUN_BATCH_TEST_EVENT: 'RunBatchTest',
};

export default amplitudeConstants;
