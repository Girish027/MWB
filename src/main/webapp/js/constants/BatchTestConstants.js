
const PREDICTED_RUI = 'Predicted RUI';
const batchTestConstants = {
  DEFAULT_PAGE_SIZE: 100,

  // Filter Related
  TRANSCRIPTIONS_FILTER_NAME: 'Transcriptions',
  TRANSCRIPTIONS_FILTER_LIST: ['All', 'Misclassified'],
  INTENT_FILTER_NAME: 'Intent',
  FILTER_SELECTED_ALL: 'All',
  ROLLUP_INTENT_ROW_ID: 'Rollup Intent (RUI)',
  PREDICTED_RUI_1: `${PREDICTED_RUI} 1`,
  PREDICTED_RUI_2: `${PREDICTED_RUI} 2`,
  PREDICTED_RUI_3: `${PREDICTED_RUI} 3`,
  PREDICTED_RUI_4: `${PREDICTED_RUI} 4`,
  PREDICTED_RUI_5: `${PREDICTED_RUI} 5`,
};

export default batchTestConstants;
