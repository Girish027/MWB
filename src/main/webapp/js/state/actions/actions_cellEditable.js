import * as types from './types';

export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_CREATE,
  stateKey,
});

export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_REMOVE,
  stateKey,
});

export const setActiveCell = ({ stateKey, activeRowIndex, activeColumnIndex }) => ({
  type: types.CELL_EDITABLE_SET_ACTIVE_CELL,
  stateKey,
  activeRowIndex,
  activeColumnIndex,
});

export const updateValue = ({ stateKey, value }) => ({
  type: types.CELL_EDITABLE_UPDATE_VALUE,
  stateKey,
  value,
});
