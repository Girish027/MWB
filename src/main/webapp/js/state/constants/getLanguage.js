import config from 'config';
import * as english from './english';

export const getLanguage = () => {
  switch (config.lang) {
  case 'english':
  default:
    return english;
  }
};
