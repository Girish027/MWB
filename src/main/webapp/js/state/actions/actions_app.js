
import _ from 'lodash';
import getUrl, { pathKey } from 'utils/apiUrls';
import { fetchUtil } from 'utils/fetchUtils';
import goToRoute from 'utils/routeHelpers';
import Constants from 'constants/Constants';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { logAmplitudeEvent } from 'utils/amplitudeUtils';
import axios from 'axios';
import api from 'utils/api';
import { setDefaultsForAPI } from 'utils/api';
import * as types from './types';

// Note to developers:
// Both Warning and Good Messages are shown for the period of notificationInterval.
// Bad, Persistant - till user closes the notification. This should only be used sparingly.
export const callLogIngestAPI = (message, logLevel) => {
  const promise = new Promise((resolve, reject) => {
    const url = getUrl(pathKey.logIngest, { logLevel });
    return new Promise((resolve, reject) => {
      api.post({
        url,
        headers: {
          'Content-Type': 'text/plain',
        },
        data: message,
      });
    });
  });
};

export const displayBadRequestMessage = (message) => (dispatch) => {
  dispatch({
    type: types.BAD_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_ERROR);
};

export const displayWarningRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.WARNING_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_WARNING);

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayGoodRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.GOOD_REQUEST,
    value: message,
  });

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayPersistantMessage = message => ({
  type: types.PERSISTENT_REQUEST,
  value: message,
});

export const stopShowingServerMessage = () => ({
  type: types.APP_STOP_SHOWING_SERVER_MESSAGE,
});

export const updateLoginSuccess = csrfToken => ({
  type: types.LOGIN_SUCCESS,
  value: csrfToken,
});

export const updateLoginInfo = (userId, userName, userDetails) => ({
  type: types.LOGIN_INFO,
  value: {
    userId,
    userName,
    userDetails,
  },
});

export const logoutFromOktaSuccess = () => ({
  type: types.LOGOUT_OKTA,
});

export const logoutFromMWBSuccess = () => ({
  type: types.LOGOUT_MWB,
});

export const contextMenuChange = contextMenuState => ({
  type: types.CONTEXT_MENU_CHANGE,
  contextMenuState,
});

export const modalDialogChange = modalDialogState => ({
  type: types.MODAL_DIALOG_CHANGE,
  modalDialogState,
});

const actionFileRemoveRequest = ({ fileId }) => ({
  type: types.FILE_REMOVE_REQUEST,
  fileId,
});

const actionFileRemoveError = ({ fileId, error }) => ({
  type: types.FILE_REMOVE_ERROR,
  fileId,
  error,
});

const actionFileRemoveSuccess = ({ fileId }) => ({
  type: types.FILE_REMOVE_SUCCESS,
  fileId,
});

export const updateUserGroups = data => ({
  type: types.UPDATE_USER_GROUPS,
  data,
});

const onFetchFailureRemoveFile = (error, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveError({ fileId, error }));
};

const onFetchSuccessRemoveFile = (data, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveSuccess({ fileId }));
};

export const removeFile = ({ fileId }) => (dispatch, getState) => {
  dispatch(actionFileRemoveRequest({ fileId }));

  const state = getState();
  const fetchUrl = getUrl(pathKey.files, { fileId, clientId: state.header.client.id });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'delete',
    onFetchSuccess: onFetchSuccessRemoveFile,
    onFetchFailure: onFetchFailureRemoveFile,
    dispatch,
    getState,
    params: {
      fileId,
    },
  });
};

const onFetchSuccessGetUserGroups = data => (dispatch) => {
  dispatch(updateUserGroups(data));
};

export const getUserGroups = ({ csrfToken }) => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.userGroups, { csrfToken });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetUserGroups,
    dispatch,
    getState,
  });
};

export const onFetchSuccessGetCSRFToken = data => (dispatch, getState) => {
  const csrfToken = data.value;
  setDefaultsForAPI(csrfToken);
  dispatch(updateLoginSuccess(csrfToken));
  dispatch(getUserGroups({ csrfToken }));
};

export const getCSRFToken = userId => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.csrfToken);

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetCSRFToken,
    dispatch,
    getState,
    params: {
      userId,
    },
  });
};

const logoutFromMWB = () => (dispatch, getState) => {
  dispatch(logoutFromOktaSuccess());
  dispatch(logoutFromMWBSuccess());
};

export const logoutFromOkta = () => (dispatch, getState) => {
  const state = getState();
  const { oAuthLogoutURL } = state.app;
  if (oAuthLogoutURL) {
    fetch(oAuthLogoutURL, {
      method: 'DELETE',
      credentials: 'include',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    })
      .then((response) => {
        window.location.href = Constants.LOGOUT;
        switch (response.status) {
        case 204:
          logoutFromMWB();
          logAmplitudeEvent(AmplitudeConstants.LOGOUT_EVENT, state);
          break;
        case 404:
        default:
          // TODO send log for these scenarios
          break;
        }
      })
      .catch((error) => {
        // TODO send log for these scenarios
      });
  }
};

export const changeRoute = (routeName, data, history) => (dispatch, getState) => {
  const state = getState();
  const routeData = Object.assign({}, {
    client: !_.isNil(state.header.client) ? state.header.client : {
      id: 0,
      name: 'Loading...',
    },
    projectId: state.selectedProjectId,
  }, data);
  goToRoute(routeName, routeData, history);
};
