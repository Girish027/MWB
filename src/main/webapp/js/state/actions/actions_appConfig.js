import 'regenerator-runtime/runtime';

import getUrl, { pathKey } from 'utils/apiUrls';

import { fetchUtil } from 'utils/fetchUtils';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { initializeAnalytics, logAmplitudeEvent } from 'utils/amplitudeUtils';
import api from 'utils/api';
import * as types from './types';

const appConfigSuccess = data => ({
  type: types.APP_CONFIG_SUCCESS,
  data,
});

const appConfigFailure = data => ({
  type: types.APP_CONFIG_FAILURE,
  data,
});

const appConfigRequest = () => ({
  type: types.APP_CONFIG_REQUEST,
});

export const onAppConfigSuccess = data => (dispatch, getState) => {
  dispatch(appConfigSuccess(data));
  const state = getState();
  initializeAnalytics(state);
  logAmplitudeEvent(AmplitudeConstants.LOGIN_EVENT, state);
};

export const onAppConfigFailure = error => (dispatch) => {
  dispatch(appConfigFailure(error));
};

export const fetchAppConfig = () => (dispatch, getState) => {
  dispatch(appConfigRequest());
  const fetchUrl = getUrl(pathKey.appConfig);

  api.get({
    url: fetchUrl,
    onApiSuccess: onAppConfigSuccess,
    onApiError: onAppConfigFailure,
    dispatch,
    getState,
  });
};
