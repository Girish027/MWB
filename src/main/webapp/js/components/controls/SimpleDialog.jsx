import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from '@tfs/ui-components';
import * as actionsApp from 'state/actions/actions_app';
import TaggerModal from 'components/controls/TaggerModal';
import store from 'state/configureStore';

export default class SimpleDialog extends Component {
  constructor(props) {
    super(props);
    this.props = props;

    this.onClickOk = this.onClickOk.bind(this);
    this.onClickCancel = this.onClickCancel.bind(this);
  }

  onClickCancel() {
    if (typeof this.props.onCancel === 'function') {
      this.props.onCancel();
    } else {
      store.dispatch(actionsApp.modalDialogChange(null));
    }
  }

  onClickOk() {
    if (typeof this.props.onOk === 'function') {
      this.props.onOk();
    } else {
      store.dispatch(actionsApp.modalDialogChange(null));
    }
  }


  getActions() {
    let { actions, ok, cancel } = this.props;

    if (!actions) {
      actions = [SimpleDialog.ACTION_OK, SimpleDialog.ACTION_CANCEL];
    }

    const ret = [];
    if (actions.indexOf(SimpleDialog.ACTION_CANCEL) != -1) {
      ret.push(
        <Button
          type="flat"
          name="cancel"
          key="Cancel"
          onClick={this.onClickCancel}
          data-qa="cancel"
          styleOverride={{
            ':focus': {
              outline: 'none',
            },
          }}
        >
Cancel
        </Button>,
      );
    }
    if (actions.indexOf(SimpleDialog.ACTION_OK) != -1) {
      ret.push(<Button
        name="yes"
        key="Yes"
        onClick={this.onClickOk}
        data-qa="ok"
      >
OK
      </Button>);
    }
    return ret;
  }

  render() {
    const { header, message, className } = this.props;

    return (
      <TaggerModal
        id="AddDatasetDialog"
        header={header || 'Are you sure?'}
        content={message || ''}
        actions={this.getActions()}
        className={`SimpleDialog${className ? ` ${className}` : ''}`}
      />
    );
  }
}

SimpleDialog.ACTION_OK = 'ACTION_OK';
SimpleDialog.ACTION_CANCEL = 'ACTION_CANCEL';
SimpleDialog.propTypes = {
  header: PropTypes.string,
  message: PropTypes.oneOfType([PropTypes.node, PropTypes.func, PropTypes.string]),
  ok: PropTypes.string,
  onOk: PropTypes.func,
  cancel: PropTypes.string,
  onCancel: PropTypes.func,
  actions: PropTypes.array,
  className: PropTypes.string,
};
