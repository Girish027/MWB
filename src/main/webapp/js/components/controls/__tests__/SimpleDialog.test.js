import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as actionsApp from 'state/actions/actions_app';
import { getLanguage } from 'state/constants/getLanguage';
import SimpleDialog from 'components/controls/SimpleDialog';


describe('<SimpleDialog />', () => {
  let wrapper;

  const props = {
    header: 'Apply Suggested Intents',
  };

  describe('Creating an instance with no props', () => {
    beforeEach(() => {
      wrapper = shallow(<SimpleDialog />);
    });
    test('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });

    test('Snapshots should render correctly', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Creating an instance with test props', () => {
    beforeEach(() => {
      wrapper = mount(<SimpleDialog
        {...props}
      />);
    });

    test('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });

    test('Snapshots should render correctly', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality', () => {
    beforeAll(() => {
      props.dispatch = jest.fn();
      props.onCancel = jest.fn();
      props.onOk = jest.fn();
      actionsApp.modalDialogChange = jest.fn(() => 'called modalDialogChange');
    });

    beforeEach(() => {
      wrapper = shallow(<SimpleDialog
        {...props}
      />);
    });

    describe('onClickCancel:', () => {
      test('should call onClickCancel click handler', () => {
        const props = {
          header: 'Simple Dialog',
          onCancel: () => {},
          dispatch: () => {},
        };

        wrapper.instance().onClickCancel();
        wrapper = shallow(<SimpleDialog
          {...props}
        />);
        expect(props.onCancel).toHaveBeenCalled;
      });
    });

    describe('onClickOk:', () => {
      test('should call onClickOk click handler', () => {
        const props = {
          header: 'Simple Dialog',
          onOk: () => {},
          dispatch: () => {},
        };

        wrapper.instance().onClickOk();
        wrapper = shallow(<SimpleDialog
          {...props}
        />);
        expect(props.onOk).toHaveBeenCalled;
      });
    });
  });
});
