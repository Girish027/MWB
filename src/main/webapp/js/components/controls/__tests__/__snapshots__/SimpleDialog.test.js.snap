// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleDialog /> Creating an instance with no props Snapshots should render correctly 1`] = `
<TaggerModal
  actions={
    Array [
      <withTheme(Button)
        data-qa="cancel"
        name="cancel"
        onClick={[Function]}
        styleOverride={
          Object {
            ":focus": Object {
              "outline": "none",
            },
          }
        }
        type="flat"
      >
        Cancel
      </withTheme(Button)>,
      <withTheme(Button)
        data-qa="ok"
        name="yes"
        onClick={[Function]}
      >
        OK
      </withTheme(Button)>,
    ]
  }
  className="SimpleDialog"
  content=""
  header="Are you sure?"
  id="AddDatasetDialog"
/>
`;

exports[`<SimpleDialog /> Creating an instance with test props Snapshots should render correctly 1`] = `
<SimpleDialog
  header="Apply Suggested Intents"
>
  <TaggerModal
    actions={
      Array [
        <withTheme(Button)
          data-qa="cancel"
          name="cancel"
          onClick={[Function]}
          styleOverride={
            Object {
              ":focus": Object {
                "outline": "none",
              },
            }
          }
          type="flat"
        >
          Cancel
        </withTheme(Button)>,
        <withTheme(Button)
          data-qa="ok"
          name="yes"
          onClick={[Function]}
        >
          OK
        </withTheme(Button)>,
      ]
    }
    className="SimpleDialog"
    content=""
    header="Apply Suggested Intents"
    id="AddDatasetDialog"
  >
    <Layer
      align="center"
      className="TaggerControl TaggerModal SimpleDialog"
      closer={false}
      id="TaggerModal"
    >
      <span
        style={
          Object {
            "display": "none",
          }
        }
      />
    </Layer>
  </TaggerModal>
</SimpleDialog>
`;
